### PR TITLE
Expose CIRE knobs

### PR DESCRIPTION
**Pull request by FabioLuporini**

### CODE REVIEW

File: devito/core/cpu.py [hunk 1]
```diff
@@ -35,6 +35,19 @@ class CPU64NoopOperator(OperatorCore):
...
+    Minimum operation count of a Dimension-invariant aliasing expression to be
+    optimized away. Dimension-invariant aliases are lifted outside of one or more
+    invariant loop(s), so they require tensor temporaries that can be potentially
+    very large (e.g., the whole domain in the case of time-invariant aliases).
+    """
+
+    CIRE_MINCOST_SOPS = 10
+    """
+    Minimum operation count of a sum-of-product aliasing expression to be optimized away.
+    """
```
Comment (mloubout) [inline devito/core/cpu.py:49]: can't we move these defaults to `OperatorCore` instead, unnecessary reapeat here.

File: devito/core/gpu_openmp.py [hunk 2]
```diff
@@ -230,6 +230,19 @@ class DeviceOpenMPNoopOperator(OperatorCore):
...
+    Minimum operation count of a Dimension-invariant aliasing expression to be
+    optimized away. Dimension-invariant aliases are lifted outside of one or more
+    invariant loop(s), so they require tensor temporaries that can be potentially
+    very large (e.g., the whole domain in the case of time-invariant aliases).
+    """
+
+    CIRE_MINCOST_SOPS = 10
+    """
+    Minimum operation count of a sum-of-product aliasing expression to be optimized away.
+    """
```
Comment (mloubout) [inline devito/core/gpu_openmp.py:244]: same

File: devito/passes/clusters/aliases.py [hunk 3]
```diff
@@ -80,22 +65,32 @@ def cire(cluster, template, mode, options, platform):
...
-    assert all(i > 0 for i in options['cire-repeats'].values())
-
     # Relevant options
     min_storage = options['min-storage']
+    min_cost = options['cire-mincost']
+    repeats = options['cire-repeats']
+
+    # Sanity checks
+    assert mode in ['invariants', 'sops']
+    assert all(i > 0 for i in repeats.values())
```
Comment (mloubout) [inline devito/passes/clusters/aliases.py:75]: do we want to allow `0` to get a zero optization operator? just a thought

File: devito/core/cpu.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/core/cpu.py:49]: no, they will change at some point for GPUs. And there already are a few differences, the CIRE_REPEATS_INV, for example, is 2 in the GPU one. As you may guess, I thought carefully about centralizing this, but it was growing into a big mess, with partial overrides in the individual operators, so I gave up. I prefer the **root** Operator classes (the CPU root, and the GPU root) to define them explicitly

File: devito/passes/clusters/aliases.py (hunk 3 above)
Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:75]: I could allow it... but if one really wants to disable it, the way to go is to use a CustomOperator as we do in some tests btw

File: devito/core/cpu.py (hunk 1 above)
Comment (mloubout) [reply devito/core/cpu.py:49]: Ok fine with me

File: devito/passes/clusters/aliases.py (hunk 3 above)
Comment (mloubout) [reply devito/passes/clusters/aliases.py:75]: I get that but this would be an "easier" way to have quick access to a basic noop operator.

Comment (FabioLuporini) [reply devito/passes/clusters/aliases.py:75]: OK, fine by me. Do you mind if I do that in the PR that sits right on top of this? it's the patch for the bug we discovered yesterday. So we don't have to respin CI, rebase, etc

Comment (mloubout) [reply devito/passes/clusters/aliases.py:75]: Completely fine yes. Played with that other one and looks like it fixes the issue so can probably open the PR (that big try/except not sure but does the job).

**Pull request merged**